packages/mobile/src/exchange/ExchangeTradeScreen.tsx

```diff
@@ -73,41 +82,46 @@ type OwnProps = StackScreenProps<StackParamList, Screens.ExchangeTradeScreen>
 type Props = StateProps & DispatchProps & WithTranslation & OwnProps
 
 const mapStateToProps = (state: RootState): StateProps => ({
-  exchangeRatePair: state.exchange.exchangeRatePair,
+  exchangeRates: exchangeRatesSelector(state),
   error: errorSelector(state),
   localCurrencyCode: getLocalCurrencyCode(state),
-  localCurrencyExchangeRate: getLocalCurrencyToDollarsExchangeRate(state),
+  balances: balancesSelector(state),
+  localCurrencyExchangeRates: localCurrencyExchangeRatesSelector(state),
+  defaultCurrency: defaultCurrencySelector(state),
 })
 
 export class ExchangeTradeScreen extends React.Component<Props, State> {
   state: State = {
-    inputToken: Currency.Celo,
-    makerToken: Currency.Dollar,
-    makerTokenAvailableBalance: '',
+    inputToken: InputToken.Celo,
+    buyCelo: true,
     inputAmount: '', // Raw amount entered, can be cGLD, cUSD or local currency
     exchangeRateInfoDialogVisible: false,
+    transferCurrency: Currency.Dollar,
   }
 
   componentDidMount() {
-    this.getMakerTokenPropertiesFromNavProps()
+    this.getSellTokenPropertiesFromNavProps()
   }
 
-  getMakerTokenPropertiesFromNavProps = () => {
-    const { makerToken, makerTokenBalance } = this.props.route.params.makerTokenDisplay
-
-    if (!makerToken) {
-      throw new Error('Maker token missing from nav props')
-    }
-
-    if (!makerTokenBalance) {
-      throw new Error('Maker token balance missing from nav props')
-    }
+  getSellTokenPropertiesFromNavProps = () => {
+    const { buyCelo } = this.props.route.params
+    const { defaultCurrency, balances } = this.props
 
     this.setState({
-      makerToken,
-      makerTokenAvailableBalance: makerTokenBalance,
+      buyCelo,
+      transferCurrency: defaultCurrency,
     })
-    this.props.fetchExchangeRate(makerToken, new BigNumber(makerTokenBalance))
+
+    const balanceIsMissing = Object.keys(CURRENCIES).some(
+      (currency) => balances[currency as Currency] === null
+    )
+    if (balanceIsMissing) {
+      this.props.showError(ErrorMessages.FETCH_FAILED)
+      navigateBack()
+      return
+    }
```

Comment:
Feels weird to have this logic happen on mount.
Why are we navigating to this screen if we don't have the necessary data?

Reply:
We're not, but it's either this or use a `!` for the balance variable. I just felt this was more defensive in case we do by error end up here, but I'm open to doing something else 🤷‍♂️  In the meantime I added a log saying 'Should never happen'